### PR TITLE
[SDK-75][Node.js SDK] Decouple validations from constructor

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [ ] https://incountry.atlassian.net/browse/SDK-75

Code Quality (Before submitting PR)
------------
- [ ] Ensure code compiles and passes Sonar Lint.
- [ ] Do your best to ensure integration tests pass. If they don't then...
- [ ] Remove, ignore, or mark pending any brittle tests.
- [ ] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: tests/unit/in-crypt-tests.js [hunk 1]
```diff
@@ -133,36 +133,45 @@ describe('InCrypt', function () {
...
           }],
         ].map((configs) => {
           const incrypt = new InCrypt(secretKeyAccessor);
-          return expect(incrypt.initialize(configs)).to.be.rejectedWith(StorageClientError, '<CustomEncryptionConfigs>');
+          return expect(() => incrypt.setCustomEncryption(configs)).to.throw(StorageClientError, '<CustomEncryptionConfigs>');
         }),
       );
     });
 
+    it('should check encrypt and decrypt', () => {
```
Comment (artemgolozubov) [inline tests/unit/in-crypt-tests.js:171]: Do we really need this empty test?

File: lib/in-crypt.js [hunk 2]
```diff
@@ -81,6 +76,57 @@ class InCrypt {
...
+            throw new StorageCryptoError(`${context} decrypted data doesn't match the original input`);
+          }
+        } catch (e) {
+          throw new StorageCryptoError(`${context} ${e.message}`);
+        }
+      }))
+        .flat()
+        .map(reflect),
+    )
+      .then((results) => {
```
Comment (artemgolozubov) [inline lib/in-crypt.js:121]: Looks like too long chain IMHO. I'd prefer to wait for results firstly and then process it in a separate block.

File: README.md [hunk 3]
```diff
@@ -34,6 +34,25 @@ const storage = await createStorage({
 
 `apiKey` and `environmentId` can be fetched from your dashboard on `Incountry` site.
 
+
+Otherwise you can create an instance of `Storage` class and run all async checks by yourself (or not run by you own risk!)
```
Comment (RommeroInCountry) [inline README.md:38]: at your own risk

File: README.md [hunk 4]
```diff
@@ -34,6 +34,25 @@ const storage = await createStorage({
...
+  environmentId: 'ENVIRONMENT_ID',
+  endpoint: 'INC_URL',
+  encrypt: true,
+  getSecrets: () => '',
+});
+
+await storage.validate();
+```
+
+`validate` method fetches secret data using `GetSecretsCallback` and validates it. If custom encryption configs had been provided they would be checked also with all matching secrets.
```
Comment (RommeroInCountry) [inline README.md:53]: if custom encryption configs were provided they would also be checked...

File: lib/in-crypt.js [hunk 5]
```diff
@@ -81,6 +76,57 @@ class InCrypt {
     }
   }
 
+  async validateCustomEncryption() {
```
Comment (RommeroInCountry) [inline lib/in-crypt.js:79]: let's separate it in utils or smth. No need to have too much non-encryption-releated code in Crypto file.

File: README.md [hunk 6]
```diff
@@ -50,7 +50,7 @@ const storage = new Storage({
 await storage.validate();
 ```
 
-`validate` method fetches secret data using `GetSecretsCallback` and validates it. If custom encryption configs had been provided they would be checked also with all matching secrets.
+`validate` method fetches secret data using `GetSecretsCallback` and validates it. If custom encryption configs were provided they would also be checked also with all matching secrets.
```
Comment (RommeroInCountry) [inline README.md:53]: too many alsos :D

**Pull request merged**